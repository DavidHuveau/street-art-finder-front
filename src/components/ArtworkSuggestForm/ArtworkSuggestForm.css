
.Suggest {
  border: solid 1px#ddd;
  border-radius: 4px;
  margin-bottom: 20px;
}

.Suggest .card-header {
  text-align: center;
  padding: 10px 15px;
}

.Suggest .card-body {
  padding-left: 16px;
}

.Suggest .form-row {
  margin-right: 0px !important;
  margin-left: 0px !important;
}

.Suggest .form-label {
  font-weight: 700;
}

.Suggest #input-zipcode {
  margin-right: 16px;
}


/* @media screen and (min-width: 425px) {
  #input-city {
    margin-left: 16px;
  }
} */
