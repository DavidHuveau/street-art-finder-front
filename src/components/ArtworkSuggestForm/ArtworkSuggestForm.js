import React, { Component, Fragment } from "react";
import "./ArtworkSuggestForm.css";
import { Form, Button, Spinner, Card, Container } from "react-bootstrap";

class ArtworkSuggestForm extends Component {
  constructor(props) {
    super(props);

    ["onChangeValue", "onSubmit", "fileChangedHandler"].forEach(
      fn => (this[fn] = this[fn].bind(this))
    );

    this.state = {
      userName: "",
      adressStreet: "",
      zipCode: "",
      city: "",
      description: "",
      country: "5d8919ca6a7d8d996731b976",
      countryCode: "FR",
      selectedFile: "",
      artistName: ""
    };
  }

  componentDidUpdate(prevProps) {
    const { error, isUploading, onRequestClose } = this.props;
    const { isUploading: prevIsUploading } = prevProps;
    const isEndOfCallApiWithoutError = !isUploading && prevIsUploading;

    if (isEndOfCallApiWithoutError && !error) {
      onRequestClose();
    }
  }

  onChangeValue(e) {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  fileChangedHandler(e) {
    this.setState({
      selectedFile: e.target.files[0]
    });
  }

  onSubmit(e) {
    e.preventDefault();

    const { onSubmit } = this.props;
    onSubmit(this.state);
  }

  renderForm() {
    const { userName, adressStreet, zipCode, city, description, artistName } = this.state;

    return (
      <div className="down-lg">
        <Form.Group controlId="formGroupName">
          <Form.Label>Artist name</Form.Label>
          <Form.Control
            type="text"
            value={artistName}
            onChange={this.onChangeValue}
            name="artistName"
            required
          />
        </Form.Group>
        <Form.Group controlId="formGroupAddress">
          <Form.Label>Address</Form.Label>
          <Form.Control
            type="text"
            placeholder="Street address..."
            value={adressStreet}
            onChange={this.onChangeValue}
            name="adressStreet"
            required
          />
        </Form.Group>
        <Form.Row>
          <Form.Group controlId="formGroupZipCode" id="input-zipcode">
            <Form.Control
              type="text"
              placeholder="Zip code..."
              value={zipCode}
              onChange={this.onChangeValue}
              name="zipCode"
              required
            />
          </Form.Group>
          <Form.Group controlId="formGroupCity">
            <Form.Control
              type="text"
              placeholder="City..."
              value={city}
              onChange={this.onChangeValue}
              name="city"
              required
            />
          </Form.Group>
        </Form.Row>
        <Form.Group controlId="formGroupDescription">
          <Form.Label>Description</Form.Label>
          <Form.Control
            as="textarea"
            rows="3"
            value={description}
            onChange={this.onChangeValue}
            name="description"
            required
          />
        </Form.Group>
        <Form.Group controlId="formGroupName">
          <Form.Label>Photo credits</Form.Label>
          <Form.Control
            type="text"
            placeholder="Name or username..."
            value={userName}
            onChange={this.onChangeValue}
            name="userName"
            required
          />
        </Form.Group>
        <Form.Group controlId="formGroupImage">
          <Form.Label>Image</Form.Label>
          <Form.Control
            type="file"
            accept="image/*"
            onChange={this.fileChangedHandler}
            name="myFile"
            required
          />
        </Form.Group>
      </div>
    );
  }

  renderSubmitButton() {
    const { isUploading } = this.props;

    const contentButton = isUploading ? (
      <Fragment>
        <Spinner as="span" animation="border" size="sm" role="status" key="1" />
        <span key="2">Uploading...</span>
      </Fragment>
    ) : (
      "Suggest"
    );

    return (
      <div className="float-right">
        <Button variant="primary" type="submit" disabled={isUploading}>
          {contentButton}
        </Button>
      </div>
    );
  }

  render() {
    return (
      <Form onSubmit={this.onSubmit}>
        <Container style={{ marginBottom: "70px" }}>
          <Card bsPrefix="Suggest" className="down-lg">
            <Card.Header>
              <strong>Suggest</strong>
            </Card.Header>
            <Card.Body>
              <p>
                Fill in the form below. If appoved, your suggested place will
                appear on this map!
              </p>
              {this.renderForm()}
            </Card.Body>
          </Card>
          {this.renderSubmitButton()}
        </Container>
      </Form>
    );
  }
}

export default ArtworkSuggestForm;
