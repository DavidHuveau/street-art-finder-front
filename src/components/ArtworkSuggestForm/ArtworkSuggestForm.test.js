import React from "react";
import renderer from "react-test-renderer";
import ArtworkSuggestForm from "./ArtworkSuggestForm";

describe("Check Components", () => {
  it("Component ArtworkSuggestForm", () => {
    const tree = renderer
      .create(<ArtworkSuggestForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})