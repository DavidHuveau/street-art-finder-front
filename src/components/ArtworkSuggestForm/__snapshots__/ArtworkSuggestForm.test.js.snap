// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Components Component ArtworkSuggestForm 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="container"
    style={
      Object {
        "marginBottom": "70px",
      }
    }
  >
    <div
      className="down-lg Suggest"
    >
      <div
        className="card-header"
      >
        <strong>
          Suggest
        </strong>
      </div>
      <div
        className="card-body"
      >
        <p>
          Fill in the form below. If appoved, your suggested place will appear on this map!
        </p>
        <div
          className="down-lg"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupName"
            >
              Name
            </label>
            <small
              className="text-muted form-text"
            >
              Your name.
            </small>
            <input
              className="form-control"
              id="formGroupName"
              name="userName"
              onChange={[Function]}
              placeholder="Enter a name..."
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupAddress"
            >
              Address
            </label>
            <small
              className="text-muted form-text"
            >
              Address of the artwork.
            </small>
            <input
              className="form-control"
              id="formGroupAddress"
              name="adressStreet"
              onChange={[Function]}
              placeholder="Enter a street address..."
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-row"
          >
            <div
              className="form-group"
              id="input-zipcode"
            >
              <input
                className="form-control"
                id="formGroupZipCode"
                name="zipCode"
                onChange={[Function]}
                placeholder="Enter a zip code..."
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="formGroupCity"
                name="city"
                onChange={[Function]}
                placeholder="Enter a city..."
                required={true}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupDescription"
            >
              Description
            </label>
            <small
              className="text-muted form-text"
            >
              Description of the artwork.
            </small>
            <textarea
              className="form-control"
              id="formGroupDescription"
              name="description"
              onChange={[Function]}
              placeholder="Enter a description..."
              required={true}
              rows="3"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formGroupImage"
            >
              Image
            </label>
            <small
              className="text-muted form-text"
            >
              Image of the artwork.
            </small>
            <input
              accept="image/*"
              name="myFile"
              onChange={[Function]}
              required={true}
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="float-right"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Suggest
      </button>
    </div>
  </div>
</form>
`;
